tiledb/sm/array_schema/dimension.cc

```diff
@@ -222,23 +222,29 @@ const Range& Dimension::domain() const {
 void Dimension::dump(FILE* out) const {
   if (out == nullptr)
     out = stdout;
+  std::stringstream ss;
+  dump_ss(ss);
+  fprintf(out, "%s", ss.str().c_str());
```

Comment:
- No need for `fprintf`. A call to `fwrite` is all that's needed.
- Should call `fflush` at the end, since this is the top-level call.